tools/s3_quota_notice.py

```diff
@@ -0,0 +1,286 @@
+
+        self.url = "{0}{1}/".format(args.host, args.admin_entry)
+
+        if args.quota_percent.endswith("%"):
+            args.quota_percent = args.quota_percent[:-1]
+
+        self.host = args.host
+        self.insecure = args.insecure
+        self.access_key = args.access_key
+        self.secret_key = args.secret_key
+        self.quota_percent = int(args.quota_percent)
+        self.smtp_host = args.smtp_host
+        self.smtp_port = args.smtp_port
+        self.smtp_starttls = args.smtp_starttls
+        self.smtp_login = args.smtp_login
+        self.smtp_password = args.smtp_password
+        self.smtp_from = args.smtp_from
+        self.smtp_subject = args.smtp_subject
+        self.smtp_signature = args.smtp_signature
+
+        self.logger()
+
+
+    def logger(self):
+        """
+        Console Handler - for output to stdout.
+        """
+
+        root_logger = logging.getLogger()
+        root_logger.setLevel(logging.INFO)
+        log_formatter = logging.Formatter(fmt="[%(asctime)s] %(message)s",
+                                          datefmt="%a %b %d %H:%M:%S %Z %Y")
+
+        console_handler = logging.StreamHandler(sys.stdout)
+        console_handler.setFormatter(log_formatter)
+        root_logger.addHandler(console_handler)
+
+
+    def bytes_to_human_readable(self, byte, suffix="B"):
+        """
+        Convert bytes to human readable sizes.
+        """
+
+        for unit in ['', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi']:
+            if abs(byte) < 1024.0:
+                return byte, "{0}{1}".format(unit, suffix)
+            byte /= 1024.0
+        return byte, "{0}{1}".format('Yi', suffix)
+
+
+    def make_rgw_query(self, endpoint, params):
+      """
+      Generic API requester.
+      """
+
+      url = "{0}{1}{2}".format(self.url, endpoint, params)
+
+      try:
+        # Inversion of condition, when '--insecure' is defined we disable
+        # requests warning about certificate hostname mismatch.
+        if not self.insecure:
+            warnings.filterwarnings('ignore', message='Unverified HTTPS request')
+
+        response = requests.get(url, verify=self.insecure,
+                                auth=S3Auth(self.access_key,
+                                self.secret_key,
+                                self.host))
+
+        if response.status_code == requests.codes.ok:
+            return response.json()
+
+        else:
+            logging.error("RGW Error [{0}]: {1}".format(response.status_code,
+                                                        response.content.decode('utf-8')))
+
+      # DNS, connection errors, etc
+      except requests.exceptions.RequestException as e:
+          logging.error("RGW Error: {0}".format(e))
+
+
+    def get_rgw_users(self):
+        """
+        API request to get users.
+        """
+
+        rgw_users = self.make_rgw_query("user", "?list")
+
+        if rgw_users:
+            return rgw_users['keys']
```

Comment:
ceph has no `s3url/admin/user?list` uri, the correct value would be `"metadata/user"` on the first parameter instead.
what you get back is an array of the user uids, e.g. `['user1','user2']` and not a dict.

Reply:
`user?list` is a new for admin API. I've posted URL's to tracker in description.

Reply:
if `metadata/user` works for luminous mimic and nautilus, i would suggest we use that instead. If not, try to implement this functionality of listing users for all mentioned versions, or exit the script if the ceph version is not the required one. It's not good to run into weird responses while running the script.

Reply:
Indeed. Can you provide example of `metadata/user` query? I don't see any documentation for this method. Account need some read caps for this?
Thanks.

Reply:
the account needs 'metadata=read or *' caps, my first comment shows you exactly what rgw_users is after changing `user` to `metadata/user`

Reply:
Added fallback query.